PatientMonitoringSystem/PatientMonitoringSystem/Views/ModuleRowView.cs

```diff
@@ -78,7 +79,15 @@ public void UpdateCurrentReading()
 
 		public void UpdateCurrentReading(int reading)
 		{
-			CurrentReadingDisplay.Value = reading;
+			if (CurrentReadingDisplay.InvokeRequired)
+			{
+				var dlg = new SafeUpdateCurrentReading(UpdateCurrentReading);
+				CurrentReadingDisplay.Invoke(dlg, new object[] { reading });
+			}
+			else
+			{
+				CurrentReadingDisplay.Value = reading;
+			}
```

Comment:
@harrytpstudent This could be shortened to `CurrentReadingDisplay.Invoke(() => DoStuffOnUiThread())`, but the code for checking how to safely invoke something should be in the caller, and not the thing being called. For example, you could revert this change and put the following in `ModuleRowController`: `moduleRowView.Invoke(() => DoStuffOnUiThread())` It looks to me that you're making this `UpdateCurrentReading` method call itself the first time, which is a bit of an anti-pattern. Also, it looks like this code is constantly being hit, even when no bedside forms are visible, probably because the forms are all instantiated and cached when the program starts. I don't think this a good idea, because the frontend and backend should be separate, and the frontend should not drive the behaviour of the backend, i.e. you currently need this specific graphical user interface in order for the core functionality to work.